src/mempool.rs

```diff
@@ -91,10 +96,29 @@ impl Mempool {
 
     async fn handle_net_message(&mut self, command: MempoolNetMessage) {
         match command {
-            MempoolNetMessage::NewTx(tx) => {
-                info!("Got new tx {} {:?}", tx.hash(), tx);
-                self.pending_txs.push(tx.clone());
-            }
+            MempoolNetMessage::NewTx(tx) => self.on_new_tx(tx, false).await,
+        }
+    }
+
+    async fn handle_api_message(&mut self, command: RestApiMessage) {
+        match command {
+            RestApiMessage::NewTx(tx) => self.on_new_tx(tx, true).await,
+        }
+    }
+
+    async fn on_new_tx(&mut self, tx: Transaction, should_rebroadcast: bool) {
```

Comment:
I don't like boolean parameters like that, I really prefer having 2 separated functions

https://alexkondov.com/should-you-pass-boolean-to-functions/

Reply:
faisable aussi